aiida/manage/configuration/settings.py

```diff
@@ -24,9 +25,9 @@
 DEFAULT_DAEMON_DIR_NAME = 'daemon'
 DEFAULT_DAEMON_LOG_DIR_NAME = 'log'
 
-AIIDA_CONFIG_FOLDER = None
-DAEMON_DIR = None
-DAEMON_LOG_DIR = None
+AIIDA_CONFIG_FOLDER: typing.Optional[pathlib.Path] = None
```

Comment:
we are changing the type of some rather important configuration variable here.
a [quick search on github](https://github.com/search?p=2&q=AIIDA_CONFIG_FOLDER&type=Code)  shows that AIIDA_CONFIG_FOLDER is being used e.g. in aiida-vasp or aiida-pytest.

In the two examples I saw it is anyhow converted to a `Path` so I think those would not break, but I wonder whether this change is worth the risk (one could keep it a string and still use pathlib to manipulate the path inside the functions below).

If we are going to make the switch to pathlib for this variable, I would prefer if this was part of an effort to comb through aiida-core and convert all module-level variables that contain paths to `Path`s (`path.join` still gives 100+ hits on the codebase).

Reply:
I thought that since most stdlib modules support `pathlib.Path` and `str` interchangeably, this would be ok. At least for Python 3.6 but Python 3.5 is no longer supported and Python 3.6 is even on its way out. In addition, the `AIIDA_CONFIG_FOLDER` variable is not part of the public API. So all these things combined, I didn't think it was a problem. I would love to see a general replacement of all path operations within the source code to use `pathlib` but this would be a lot of work and don't think anyone will do this in one go. Instead I think we should do this progressively as code is being touched, like I do here.

Reply:
Ok - if that change is indeed very unlikely to break code in the plugins I'm fine with it.
As I mentioned the two examples I saw are not affected.